app/src/main/java/ml/bmlzootown/hydravion/card/CardPresenter.kt

```diff
@@ -132,6 +134,15 @@ class CardPresenter : Presenter() {
                         .into(image)
                 }
 
+                video.metadata?.videoDurationInSecs?.let { totalDurationSecs ->
+                    if (totalDurationSecs <= 0) {
+                        duration.isGone = true
+                    }
+                    else {
+                        duration.text = formatDuration(totalDurationSecs)
```

Comment:
You will need to set `isVisible = true` here to ensure that the previous state isn't carried over

Reply:
Oh, you're right! Let me update it now.